src/connectors/h1_collect.py

```diff
@@ -0,0 +1,316 @@
+        comment=comment,
+    )
+    db.execute(f'GRANT INSERT, SELECT ON {landing_reports_table} TO ROLE {SA_ROLE}')
+    return {
+        'newStage': 'finalized',
+        'newMessage': "HackerOne reports ingestion table created!",
+    }
+
+
+def ingest(table_name, options, dryrun=False):
+    ingest_type = (
+        'transaction' if table_name.endswith('_TRANSACTIONS_CONNECTION') else 'report'
+    )
+    landing_table = f'data.{table_name}'
+
+    timestamp = datetime.utcnow()
+    api_identifier = options['api_identifier']
+    api_token = options['api_token']
+    account_id = options['account_id']
+    program_name = options['program_name']
+
+    if ingest_type == 'transaction' and account_id:
+        try:
+            transactions = load_data(
+                f'https://api.hackerone.com/v1/programs/{account_id}/billing/transactions',
+                api_token,
+                api_identifier,
+                params=None,
+            )
+        except requests.exceptions.HTTPError as e:
+            log.error(e)
+        else:
+            db.insert(
+                landing_table,
+                dryrun=dryrun,
+                values=[
+                    {
+                        'timestamp': timestamp,
+                        'type': transaction.get('type', ''),
+                        'activity_date': get_path(
+                            transaction, 'attributes.activity_date'
+                        ),
+                        'activity_description': get_path(
+                            transaction, 'attributes.activity_description'
+                        ),
+                        'bounty_award': get_path(
+                            transaction, 'attributes.bounty_award'
+                        ),
+                        'bounty_fee': get_path(transaction, 'attributes.bounty_fee'),
+                        'debit_or_credit_amount': get_path(
+                            transaction, 'attributes.debit_or_credit_amount'
+                        ),
+                        'balance': get_path(transaction, 'attributes.balance'),
+                        'id': get_path(transaction, 'relationships.report.data.id'),
+                        'url': get_path(transaction, 'relationships.report.links.self'),
+                    }
+                    for transaction in transactions
+                ],
+            )
+    else:
+        try:
+            reports = load_data(
+                'https://api.hackerone.com/v1/reports',
+                api_token,
+                api_identifier,
+                params={'filter[program][]': program_name},
+            )
+        except requests.exceptions.HTTPError as e:
+            log.error(e)
+        else:
+            db.insert(
+                landing_table,
+                dryrun=dryrun,
+                values=[
+                    {
+                        'timestamp': timestamp,
+                        'id': report.get('id', ''),
```

Comment:
`id` col is defined as number above so I don't think you can set it to `''` without raising an error

Reply:
Is None or 0 an acceptable default to pass here then?

Reply:
I would say `None` as `0` might be an actual `id` (unlikely, but the very first email for a customer, maybe?)